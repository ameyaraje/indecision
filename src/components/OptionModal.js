import React from 'react';
import Modal from 'react-modal';

const OptionModal = (props) => {
    return (
        <Modal className="modal" isOpen={!!props.selectedOption} contentLabel="Selected Option" onRequestClose={props.clearSelectedOption} closeTimeoutMS={200}>
            <h2 className="modal__title">Selected Option</h2>
            {props.selectedOption && <p className="modal__body">{props.selectedOption}</p>}
            <button className="button" onClick={props.clearSelectedOption}>Okay</button>
        </Modal>
    );
};

export default OptionModal;