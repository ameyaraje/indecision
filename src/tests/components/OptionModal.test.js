import React from 'react';
import { shallow } from 'enzyme';

import OptionModal from '../../components/OptionModal';

test('should render OptionModal component', () => {
    const wrapper = shallow(<OptionModal />);

    expect(wrapper).toMatchSnapshot();
});