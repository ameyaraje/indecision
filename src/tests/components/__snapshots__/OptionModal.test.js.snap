// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render OptionModal component 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Selected Option"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h2
    className="modal__title"
  >
    Selected Option
  </h2>
  <button
    className="button"
  >
    Okay
  </button>
</Modal>
`;
